Library/Homebrew/cmd/update-report.rb

```diff
@@ -167,16 +170,113 @@ def migrate_legacy_cache_if_necessary
     end
   end
 
-  def link_completions_and_docs
-    return if HOMEBREW_PREFIX.to_s == HOMEBREW_REPOSITORY.to_s
+  def migrate_legacy_repository_if_necessary
+    return unless HOMEBREW_PREFIX.to_s == "/usr/local"
+    return unless HOMEBREW_REPOSITORY.to_s == "/usr/local"
+    return unless ARGV.homebrew_developer?
+
+    unless HOMEBREW_PREFIX.writable_real?
+      onoe <<-EOS.undent
+        #{HOMEBREW_PREFIX} is not writable.
+
+        You should change the ownership and permissions of #{HOMEBREW_PREFIX}
+        temporarily back to your user account so we can complete the Homebrew
+        repository migration:
+          sudo chown -R $(whoami) #{HOMEBREW_PREFIX}
+      EOS
+    end
```

Comment:
Yes, explanations would be nice, and this can be combined with my suggestion in https://github.com/Homebrew/brew/pull/963#pullrequestreview-91901.